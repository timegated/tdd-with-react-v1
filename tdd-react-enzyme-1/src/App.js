import React from 'react';
function App() {
  const [count, setCount] = React.useState(0);

  const handleDecrementCount = () => {
    setCount(count - 1);
    if(count <= 0) {
      setCount(0);
    }
  };

  const handleResetCount = () => {
    setCount(0);
  };

  return (
    <div data-test="component-app" className="App">
    <h1 data-test="component-display">
      This counter is currently&nbsp;
      <span data-test="count">{count}</span>
    </h1>
    <button
      data-test="increment-button"
      onClick={() => setCount(count + 1)}
    >
      Increment!
    </button>
    <button
      data-test="decrement-button"
      onClick={() => handleDecrementCount()}
    >
      Decrement!
    </button>
    <button
      data-test="reset-button"
      onClick={handleResetCount}
    >
      Reset!
    </button>
    </div>
  );
}

export default App;
