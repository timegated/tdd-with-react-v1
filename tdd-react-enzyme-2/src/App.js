/* eslint-disable react-hooks/exhaustive-deps */
import React from 'react';
import { useSetRecoilState } from 'recoil';
import { secretWordState } from './store/atoms';
import SecretWord from './components/SecretWord/SecretWord';

function App() {
  const secretWords = ['flower','plant','apple','elephant','rhyme'];
  const randomizedWord = secretWords[Math.floor(Math.random() * secretWords.length)];
  const setSecretWord = useSetRecoilState(secretWordState);

  React.useEffect(() => {
    setSecretWord(randomizedWord);
  }, []);

  return (
      <div className="App">
        <SecretWord />
      </div>
  );
}

export default App;
