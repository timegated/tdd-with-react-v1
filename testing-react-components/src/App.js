import Login from './components/Login';
import Counter from './components/Counter';
import './App.css';

function App() {
  return (
    <>
      <div className="App">
        <Counter />
      </div>
      <Login />
    </>
  );
}

export default App;
