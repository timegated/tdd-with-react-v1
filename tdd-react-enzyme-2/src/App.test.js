import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import { RecoilRoot } from 'recoil';
import App from './App';

Enzyme.configure({ adapter: new EnzymeAdapter() });

// Rendering our App Component in isolation
const setup = (props = {}) => {
  return mount(
    <RecoilRoot>
      <App {...props} />
    </RecoilRoot>,
  );
};

// Util function that finds the data-test attribute on the wrapper
const findDataTest = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};

describe('The Root App Component renders a SecretWord Component', () => {
  test('renders learn react link', () => {
    const wrapper = setup();
    const appComponent = findDataTest(wrapper, 'secret-word');
    expect(appComponent.length).toBe(1);
    expect(appComponent).toMatchSnapshot();
  });
});
