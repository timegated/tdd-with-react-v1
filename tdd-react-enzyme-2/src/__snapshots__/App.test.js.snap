// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Root App Component renders a SecretWord Component renders learn react link 1`] = `ReactWrapper {}`;
